src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -1769,31 +1836,25 @@ namespace nbl
 				}
 			}
 
-			/*
-			* 
-			*   TODO: IT MAY BE AN ISSUE VERY SOON!
-			* 
-				TODO: it needs hashes and better system for meta since gltf bundle may return more than one mesh
-				and each mesh may have more than one meshbuffer, so more meta as well
-			*/
-
-			auto getGlobalPipelineCount = [&]() // TODO change it
+			core::smart_refctd_ptr<CGLTFMetadata> glTFPipelineMetadata = core::make_smart_refctd_ptr<CGLTFMetadata>(globalPipelineMetadataSet.size());
 			{
-				size_t count = {};
-				for (auto& meshMeta : globalMetadataContainer)
-					for (auto& pipelineMeta : meshMeta)
-						++count;
-				return count;
-			};
-#endif
-
-			core::smart_refctd_ptr<CGLTFMetadata> glTFPipelineMetadata;// = core::make_smart_refctd_ptr<CGLTFMetadata>(getGlobalPipelineCount());
-#if 0
-			for (size_t i = 0; i < globalMetadataContainer.size(); ++i) // TODO change it 
-				for (size_t z = 0; z < globalMetadataContainer[i].size(); ++z)
-					glTFPipelineMetadata->placeMeta(z, cpuMeshes[i]->getMeshBufferVector()[z]->getPipeline(), *globalMetadataContainer[i][z]); 
-#endif
-			return SAssetBundle(std::move(glTFPipelineMetadata),cpuMeshes);
+				glTFPipelineMetadata->instances = instances;
+				glTFPipelineMetadata->skeletons = skeletons;
+				glTFPipelineMetadata->vertexJointToSkeletonJoint = vertexJointToSkeletonJoint;
+
+				auto & scene = glTFPipelineMetadata->scenes.emplace_back();
+				scene.count = glTFPipelineMetadata->instances.size();
+				scene.instanceIDs = core::make_smart_refctd_ptr<asset::ICPUBuffer>(scene.count * sizeof(CGLTFMetadata::INSTANCE_ID));
+				auto* instanceIDs = reinterpret_cast<CGLTFMetadata::INSTANCE_ID*>(scene.instanceIDs->getPointer());
+				
+				std::iota(instanceIDs, instanceIDs + scene.count, 0);
```

Comment:
WTF?

Reply:
for always-1-scene case, this will be replaced in following commit

Reply:
why are you filling the scene's instanceIDs with 0,1,2,3,4, .. ?

Reply:
corrected